lib/logstash/outputs/syslog.rb

```diff
@@ -238,7 +238,7 @@ def setup_ssl
     require "openssl"
     ssl_context = OpenSSL::SSL::SSLContext.new
     ssl_context.cert = OpenSSL::X509::Certificate.new(File.read(@ssl_cert))
-    ssl_context.key = OpenSSL::PKey::RSA.new(File.read(@ssl_key),@ssl_key_passphrase)
+    ssl_context.key = OpenSSL::PKey::read(File.read(@ssl_key),@ssl_key_passphrase)
```

Comment:
@Given that you have added all the TLS test scaffolding  in #62  do you think that, after the merge on `main` branch of that PR and rebase of this, could we add 2 tests here, one for RSA key and the other for Elliptic Cryptography here?

Reply:
Yes absolutely, let's do that.

Reply:
I needed to remove test `invalid client private key` that was previously added in TLS tests: I now realized that `OpenSSL::PKey::read()` does not raise error when given invalid PEM file, unlike `OpenSSL::PKey::RSA.new` did. Instead, it returns instance of `OpenSSL::PKey::RSA` even though `invalid.pem` clearly is not an RSA key. 

When I fixed support for EC pkey to `PKey::read()` in jruby-openssl, I did not check this. Maybe it lacks negative checks.

Reply:
I think that it worthwhile to open an issue in https://github.com/jruby/jruby-openssl repository, accepting silently an improper file is misleading.

Reply:
Maybe the difference is that the `read` method instantiate the RSA key 
https://github.com/jruby/jruby-openssl/blob/19135259f121f9c9e95ee7f0b4c830f9267680e0/src/main/java/org/jruby/ext/openssl/PKey.java#L124

instead of the `initiliaze` method 
https://github.com/jruby/jruby-openssl/blob/19135259f121f9c9e95ee7f0b4c830f9267680e0/src/main/java/org/jruby/ext/openssl/PKeyRSA.java#L223

Reply:
In my test case I used input that does not even have valid PEM block inside, so I would have expected that even PEM block parsing should fail here https://github.com/jruby/jruby-openssl/blob/19135259f121f9c9e95ee7f0b4c830f9267680e0/src/main/java/org/jruby/ext/openssl/PKey.java#L117  -> 
https://github.com/jruby/jruby-openssl/blob/19135259f121f9c9e95ee7f0b4c830f9267680e0/src/main/java/org/jruby/ext/openssl/x509store/PEMInputOutput.java#L317
Requires some additional debugging on jruby-openssl to find out what is going on there...

I've created new issue https://github.com/jruby/jruby-openssl/issues/285. I added also the CA case which I noticed earlier.

Reply:
Oks, let's see what they answer and then I'll merge this.